### ISSUE TITLE
Chapter 9 react-query API

### ISSUE THREAD

**Issue by totterman** [PacktPublishing/Learn-React-with-TypeScript-2nd-Edition#5]
I got Chapter 9 Query examples working with slightly modified react-query API calls.

App.tsx: 
```
return defer({
        posts: queryClient.fetchQuery({ queryKey: ['postsData'], queryFn: getPosts }),
      });

```
PostsPage.tsx:
  ```
const { mutate } = useMutation({
    mutationFn: savePost,
    onSuccess: (savedPost) => { ...
```

In my dependencies I have
`"@tanstack/react-query": "^5.8.4" `
Did I miss something in the project setup, or has the react-query API changed?

Thank you for a great book!

**Comment by carlrip** [comment]
Yeah spot on, some of the React Query APIs changed between v4 and v5 so your adjustments look good if you want to use v5.

**Comment by emcclellan** [comment]
Thanks for posting this.  This morning I got to this chapter and wasn't able to get the sample to work.

@carlrip Due to the api change do we now need to type the parameter inputs always or is it due to the signature changing and in the books particular use case now needing the parameters specified?
